public/javascript/application.js

```diff
@@ -69,7 +74,8 @@ var parseGraph = function(data){
         .on("dblclick", dblclick),
       circle = node
         .append("circle")
-        .attr("r", function(d) { return d.lines / max_lines * max_circle_r + 6; }),
+        .attr("r", function(d) { return d.lines / max_lines * max_circle_r + 6; })
+          .attr("class", function (d) { return d.class; }),
```

Comment:
we need to align line 78 to 77.
also  to have a class specific if the definition is circular, you can calculate the `classForCircular(d)` here instead of line 30 and 26, we can set the circular in the definition object and the relation object without transforming it to a class value and then  transform it with `classForCircular` here and that would give the same result I guess.